### ISSUE TITLE
ghc-8.4.1 derivation is broken

### ISSUE THREAD

**Issue by mpickering** [NixOS/nixpkgs#34229]
## Issue description

The build fails at a very early stage.

@deepfire 

### Steps to reproduce

```
nix-shell -p haskell.packages.ghc841.ghc
```

```
[root@nixos:~/haskell-src-exts]# nix-shell -p haskell.packages.ghc841.ghc
these derivations will be built:
  /nix/store/lzgaghhba2ci567yyfsg41392pj1rb6m-ghc-61db0b8.drv
  /nix/store/i00rsspj8fkpx37jkr18g9hvl2lsbvim-ghc-8.4.20180122.drv
building path(s) ‘/nix/store/5z011gagfisjknv0zdcs4cq1hf94zfv6-ghc-61db0b8’
exporting git://git.haskell.org/ghc.git (rev 61db0b8941cfb7ed8941ed29bdb04bd8ef3b71a5) into /nix/store/5z011gagfisjknv0zdcs4cq1hf94zfv6-ghc-61db0b8
Initialized empty Git repository in /nix/store/5z011gagfisjknv0zdcs4cq1hf94zfv6-ghc-61db0b8/.git/
fatal: The remote end hung up unexpectedly
fatal: protocol error: bad pack header
remote: Counting objects: 477132, done.        
remote: Compressing objects: 100% (97397/97397), done.        
remote: Total 477132 (delta 361660), reused 476049 (delta 360732)        
Receiving objects: 100% (477132/477132), 110.07 MiB | 10.22 MiB/s, done.
Resolving deltas: 100% (361660/361660), done.
From git://git.haskell.org/ghc
 * [new branch]            coercible                   -> origin/coercible
 * [new branch]            coloured-core               -> origin/coloured-core
 * [new branch]            context-quantification-4426 -> origin/context-quantification-4426
 * [new branch]            data-kind-syntax            -> origin/data-kind-syntax
 * [new branch]            ghc-7.10                    -> origin/ghc-7.10
 * [new branch]            ghc-7.2                     -> origin/ghc-7.2
 * [new branch]            ghc-7.4                     -> origin/ghc-7.4
 * [new branch]            ghc-7.6                     -> origin/ghc-7.6
 * [new branch]            ghc-7.8                     -> origin/ghc-7.8
 * [new branch]            ghc-8.0                     -> origin/ghc-8.0
 * [new branch]            ghc-8.2                     -> origin/ghc-8.2
 * [new branch]            ghc-8.4                     -> origin/ghc-8.4
 * [new branch]            ghc-july                    -> origin/ghc-july
 * [new branch]            ghc-lwc2                    -> origin/ghc-lwc2
 * [new branch]            ghc-new-flavor              -> origin/ghc-new-flavor
 * [new branch]            ghc-quick3                  -> origin/ghc-quick3
 * [new branch]            ghc-spj                     -> origin/ghc-spj
 * [new branch]            ghc-validate                -> origin/ghc-validate
 * [new branch]            late-lam-lift               -> origin/late-lam-lift
 * [new branch]            local-gc                    -> origin/local-gc
 * [new branch]            master                      -> origin/master
 * [new branch]            monad-comp                  -> origin/monad-comp
 * [new branch]            no-pred-ty                  -> origin/no-pred-ty
 * [new branch]            sdoc                        -> origin/sdoc
 * [new branch]            supercompiler               -> origin/supercompiler
 * [new branch]            tc-untouchables             -> origin/tc-untouchables
 * [new branch]            th-new-7.6                  -> origin/th-new-7.6
 * [new branch]            wip/14691                   -> origin/wip/14691
 * [new branch]            wip/8144                    -> origin/wip/8144
 * [new branch]            wip/D2899                   -> origin/wip/D2899
 * [new branch]            wip/D3050                   -> origin/wip/D3050
 * [new branch]            wip/D3646                   -> origin/wip/D3646
 * [new branch]            wip/D4145                   -> origin/wip/D4145
 * [new branch]            wip/D4170                   -> origin/wip/D4170
 * [new branch]            wip/D694                    -> origin/wip/D694
 * [new branch]            wip/GenericsMetaData        -> origin/wip/GenericsMetaData
 * [new branch]            wip/GenericsMetaData2       -> origin/wip/GenericsMetaData2
 * [new branch]            wip/GrowableAST             -> origin/wip/GrowableAST
 * [new branch]            wip/T10071                  -> origin/wip/T10071
 * [new branch]            wip/T10246                  -> origin/wip/T10246
 * [new branch]            wip/T10527                  -> origin/wip/T10527
 * [new branch]            wip/T10527-2                -> origin/wip/T10527-2
 * [new branch]            wip/T10613                  -> origin/wip/T10613
 * [new branch]            wip/T10803                  -> origin/wip/T10803
 * [new branch]            wip/T10832-generalised-injectivity -> origin/wip/T10832-generalised-injectivity
 * [new branch]            wip/T10858                  -> origin/wip/T10858
 * [new branch]            wip/T10874                  -> origin/wip/T10874
 * [new branch]            wip/T10918                  -> origin/wip/T10918
 * [new branch]            wip/T11028                  -> origin/wip/T11028
 * [new branch]            wip/T11028-2                -> origin/wip/T11028-2
 * [new branch]            wip/T11067                  -> origin/wip/T11067
 * [new branch]            wip/T11168                  -> origin/wip/T11168
 * [new branch]            wip/T11179                  -> origin/wip/T11179
 * [new branch]            wip/T11224                  -> origin/wip/T11224
 * [new branch]            wip/T11970                  -> origin/wip/T11970
 * [new branch]            wip/T12105                  -> origin/wip/T12105
 * [new branch]            wip/T12144                  -> origin/wip/T12144
 * [new branch]            wip/T12357                  -> origin/wip/T12357
 * [new branch]            wip/T12357-built-in-map     -> origin/wip/T12357-built-in-map
 * [new branch]            wip/T12357-concat           -> origin/wip/T12357-concat
 * [new branch]            wip/T12357-unpack           -> origin/wip/T12357-unpack
 * [new branch]            wip/T12364                  -> origin/wip/T12364
 * [new branch]            wip/T12618                  -> origin/wip/T12618
 * [new branch]            wip/T12626                  -> origin/wip/T12626
 * [new branch]            wip/T12819                  -> origin/wip/T12819
 * [new branch]            wip/T12919                  -> origin/wip/T12919
 * [new branch]            wip/T13255-spj              -> origin/wip/T13255-spj
 * [new branch]            wip/T13351                  -> origin/wip/T13351
 * [new branch]            wip/T13594                  -> origin/wip/T13594
 * [new branch]            wip/T13644                  -> origin/wip/T13644
 * [new branch]            wip/T13719                  -> origin/wip/T13719
 * [new branch]            wip/T13861                  -> origin/wip/T13861
 * [new branch]            wip/T13904                  -> origin/wip/T13904
 * [new branch]            wip/T14068                  -> origin/wip/T14068
 * [new branch]            wip/T14152                  -> origin/wip/T14152
 * [new branch]            wip/T14152cse               -> origin/wip/T14152cse
 * [new branch]            wip/T14289                  -> origin/wip/T14289
 * [new branch]            wip/T14373                  -> origin/wip/T14373
 * [new branch]            wip/T14529                  -> origin/wip/T14529
 * [new branch]            wip/T14626                  -> origin/wip/T14626
 * [new branch]            wip/T14677                  -> origin/wip/T14677
 * [new branch]            wip/T14688                  -> origin/wip/T14688
 * [new branch]            wip/T3384                   -> origin/wip/T3384
 * [new branch]            wip/T4404                   -> origin/wip/T4404
 * [new branch]            wip/T4879                   -> origin/wip/T4879
 * [new branch]            wip/T5084                   -> origin/wip/T5084
 * [new branch]            wip/T5462                   -> origin/wip/T5462
 * [new branch]            wip/T7704                   -> origin/wip/T7704
 * [new branch]            wip/T7860                   -> origin/wip/T7860
 * [new branch]            wip/T8545-ghc-7.8           -> origin/wip/T8545-ghc-7.8
 * [new branch]            wip/T8592                   -> origin/wip/T8592
 * [new branch]            wip/T8776                   -> origin/wip/T8776
 * [new branch]            wip/T8959                   -> origin/wip/T8959
 * [new branch]            wip/T8995-level-generalisation -> origin/wip/T8995-level-generalisation
 * [new branch]            wip/T9020                   -> origin/wip/T9020
 * [new branch]            wip/T9136                   -> origin/wip/T9136
 * [new branch]            wip/T9233                   -> origin/wip/T9233
 * [new branch]            wip/T9281                   -> origin/wip/T9281
 * [new branch]            wip/T9388                   -> origin/wip/T9388
 * [new branch]            wip/T9417                   -> origin/wip/T9417
 * [new branch]            wip/T9889                   -> origin/wip/T9889
 * [new branch]            wip/T9953                   -> origin/wip/T9953
 * [new branch]            wip/T9968                   -> origin/wip/T9968
 * [new branch]            wip/Txxxxx                  -> origin/wip/Txxxxx
 * [new branch]            wip/aarch64-regd            -> origin/wip/aarch64-regd
 * [new branch]            wip/alexbiehl               -> origin/wip/alexbiehl
 * [new branch]            wip/all-inlinable           -> origin/wip/all-inlinable
 * [new branch]            wip/all-inlinable-head      -> origin/wip/all-inlinable-head
 * [new branch]            wip/angerman/llvmng         -> origin/wip/angerman/llvmng
 * [new branch]            wip/annotate-core           -> origin/wip/annotate-core
 * [new branch]            wip/az-anndotdot            -> origin/wip/az-anndotdot
 * [new branch]            wip/better-machine-readable-stats -> origin/wip/better-machine-readable-stats
 * [new branch]            wip/binary-bytestring       -> origin/wip/binary-bytestring
 * [new branch]            wip/cabal-head-updates      -> origin/wip/cabal-head-updates
 * [new branch]            wip/cabal-parsec            -> origin/wip/cabal-parsec
 * [new branch]            wip/cbv-conv-thunk          -> origin/wip/cbv-conv-thunk
 * [new branch]            wip/cheap-build             -> origin/wip/cheap-build
 * [new branch]            wip/check-uniques-fix       -> origin/wip/check-uniques-fix
 * [new branch]            wip/circleci                -> origin/wip/circleci
 * [new branch]            wip/circleci-ben            -> origin/wip/circleci-ben
 * [new branch]            wip/cmmsink                 -> origin/wip/cmmsink
 * [new branch]            wip/common-context          -> origin/wip/common-context
 * [new branch]            wip/cross-constr-cse        -> origin/wip/cross-constr-cse
 * [new branch]            wip/cross-spec-constr       -> origin/wip/cross-spec-constr
 * [new branch]            wip/custom-type-errors      -> origin/wip/custom-type-errors
 * [new branch]            wip/dfeuer-T13397           -> origin/wip/dfeuer-T13397
 * [new branch]            wip/dfeuer-interleave-mvars -> origin/wip/dfeuer-interleave-mvars
 * [new branch]            wip/dfeuer-interleave-null  -> origin/wip/dfeuer-interleave-null
 * [new branch]            wip/dfeuer-less-hammer      -> origin/wip/dfeuer-less-hammer
 * [new branch]            wip/discount-fv             -> origin/wip/discount-fv
 * [new branch]            wip/dph-fix                 -> origin/wip/dph-fix
 * [new branch]            wip/dup_app                 -> origin/wip/dup_app
 * [new branch]            wip/embelleshed-rdr         -> origin/wip/embelleshed-rdr
 * [new branch]            wip/erikd-build             -> origin/wip/erikd-build
 * [new branch]            wip/erikd/heapview          -> origin/wip/erikd/heapview
 * [new branch]            wip/erikd/linker            -> origin/wip/erikd/linker
 * [new branch]            wip/erikd/rts               -> origin/wip/erikd/rts
 * [new branch]            wip/ermsb                   -> origin/wip/ermsb
 * [new branch]            wip/exprArity               -> origin/wip/exprArity
 * [new branch]            wip/ext-solver              -> origin/wip/ext-solver
 * [new branch]            wip/faster-stats            -> origin/wip/faster-stats
 * [new branch]            wip/float-join-points       -> origin/wip/float-join-points
 * [new branch]            wip/foldl                   -> origin/wip/foldl
 * [new branch]            wip/gadtpm-prefix-tree      -> origin/wip/gadtpm-prefix-tree
 * [new branch]            wip/generics-flip           -> origin/wip/generics-flip
 * [new branch]            wip/generics-propeq         -> origin/wip/generics-propeq
 * [new branch]            wip/generics-propeq-conservative -> origin/wip/generics-propeq-conservative
 * [new branch]            wip/ggreif                  -> origin/wip/ggreif
 * [new branch]            wip/ghc-7.10-with-timings   -> origin/wip/ghc-7.10-with-timings
 * [new branch]            wip/ghc-8.0-det             -> origin/wip/ghc-8.0-det
 * [new branch]            wip/ghc-pkg-locking         -> origin/wip/ghc-pkg-locking
 * [new branch]            wip/ghc710-aix              -> origin/wip/ghc710-aix
 * [new branch]            wip/ghci-staticptrs         -> origin/wip/ghci-staticptrs
 * [new branch]            wip/hadrian                 -> origin/wip/hadrian
 * [new branch]            wip/hasfield                -> origin/wip/hasfield
 * [new branch]            wip/high_memory_usage       -> origin/wip/high_memory_usage
 * [new branch]            wip/impredicativity         -> origin/wip/impredicativity
 * [new branch]            wip/inlinable-bug           -> origin/wip/inlinable-bug
 * [new branch]            wip/interleave-rw           -> origin/wip/interleave-rw
 * [new branch]            wip/jenkins                 -> origin/wip/jenkins
 * [new branch]            wip/js-hoopl-cleanup        -> origin/wip/js-hoopl-cleanup
 * [new branch]            wip/kavon-nosplit-llvm      -> origin/wip/kavon-nosplit-llvm
 * [new branch]            wip/lazy-instance-matching  -> origin/wip/lazy-instance-matching
 * [new branch]            wip/lexer-warnings          -> origin/wip/lexer-warnings
 * [new branch]            wip/libdw-prof              -> origin/wip/libdw-prof
 * [new branch]            wip/libdw-unwind            -> origin/wip/libdw-unwind
 * [new branch]            wip/llf                     -> origin/wip/llf
 * [new branch]            wip/llvm-debug-info         -> origin/wip/llvm-debug-info
 * [new branch]            wip/merge                   -> origin/wip/merge
 * [new branch]            wip/mpickering-unfolding-discounts -> origin/wip/mpickering-unfolding-discounts
 * [new branch]            wip/mpickering/rn-env       -> origin/wip/mpickering/rn-env
 * [new branch]            wip/nested-cpr              -> origin/wip/nested-cpr
 * [new branch]            wip/new-flatten-skolems-Aug14 -> origin/wip/new-flatten-skolems-Aug14
 * [new branch]            wip/new-flatten-skolems-Oct14 -> origin/wip/new-flatten-skolems-Oct14
 * [new branch]            wip/new-tree-one-param-2    -> origin/wip/new-tree-one-param-2
 * [new branch]            wip/nfs-locking             -> origin/wip/nfs-locking
 * [new branch]            wip/orf                     -> origin/wip/orf
 * [new branch]            wip/orf-2017                -> origin/wip/orf-2017
 * [new branch]            wip/orf-new                 -> origin/wip/orf-new
 * [new branch]            wip/orf-reboot              -> origin/wip/orf-reboot
 * [new branch]            wip/pattern-synonym-backport -> origin/wip/pattern-synonym-backport
 * [new branch]            wip/pattern-synonym-sig-backport -> origin/wip/pattern-synonym-sig-backport
 * [new branch]            wip/pattern-synonyms        -> origin/wip/pattern-synonyms
 * [new branch]            wip/perf-testsuite          -> origin/wip/perf-testsuite
 * [new branch]            wip/pretty                  -> origin/wip/pretty
 * [new branch]            wip/pretty32                -> origin/wip/pretty32
 * [new branch]            wip/prettyprinter           -> origin/wip/prettyprinter
 * [new branch]            wip/ptext-sLit-cleanup      -> origin/wip/ptext-sLit-cleanup
 * [new branch]            wip/rae                     -> origin/wip/rae
 * [new branch]            wip/rae-kind-gadts          -> origin/wip/rae-kind-gadts
 * [new branch]            wip/raiseIO-conservative    -> origin/wip/raiseIO-conservative
 * [new branch]            wip/recurs-compat           -> origin/wip/recurs-compat
 * [new branch]            wip/redundant-constraints   -> origin/wip/redundant-constraints
 * [new branch]            wip/remove-cabal-dep        -> origin/wip/remove-cabal-dep
 * [new branch]            wip/revert-ttg-2017-11-20   -> origin/wip/revert-ttg-2017-11-20
 * [new branch]            wip/runtime-rep             -> origin/wip/runtime-rep
 * [new branch]            wip/rwbarton-D1259          -> origin/wip/rwbarton-D1259
 * [new branch]            wip/rwbarton-D2992          -> origin/wip/rwbarton-D2992
 * [new branch]            wip/rwbarton-D3164          -> origin/wip/rwbarton-D3164
 * [new branch]            wip/rwbarton-D3516          -> origin/wip/rwbarton-D3516
 * [new branch]            wip/rwbarton-O2             -> origin/wip/rwbarton-O2
 * [new branch]            wip/rwbarton-align4         -> origin/wip/rwbarton-align4
 * [new branch]            wip/rwbarton-biniface       -> origin/wip/rwbarton-biniface
 * [new branch]            wip/rwbarton-dep-finsts     -> origin/wip/rwbarton-dep-finsts
 * [new branch]            wip/rwbarton-dump           -> origin/wip/rwbarton-dump
 * [new branch]            wip/rwbarton-float-unboxed  -> origin/wip/rwbarton-float-unboxed
 * [new branch]            wip/rwbarton-gold           -> origin/wip/rwbarton-gold
 * [new branch]            wip/rwbarton-jp-nolint      -> origin/wip/rwbarton-jp-nolint
 * [new branch]            wip/rwbarton-large-tuple    -> origin/wip/rwbarton-large-tuple
 * [new branch]            wip/rwbarton-minusInteger   -> origin/wip/rwbarton-minusInteger
 * [new branch]            wip/rwbarton-no-core-lint   -> origin/wip/rwbarton-no-core-lint
 * [new branch]            wip/rwbarton-no-core-lint-after-jp -> origin/wip/rwbarton-no-core-lint-after-jp
 * [new branch]            wip/rwbarton-no-core-lint-before-jp -> origin/wip/rwbarton-no-core-lint-before-jp
 * [new branch]            wip/rwbarton-revert-splitter -> origin/wip/rwbarton-revert-splitter
 * [new branch]            wip/rwbarton-seq-unf        -> origin/wip/rwbarton-seq-unf
 * [new branch]            wip/rwbarton-simplify       -> origin/wip/rwbarton-simplify
 * [new branch]            wip/rwbarton-split-sections -> origin/wip/rwbarton-split-sections
 * [new branch]            wip/rwbarton-supports-split-sections -> origin/wip/rwbarton-supports-split-sections
 * [new branch]            wip/rwbarton-tiny-tables    -> origin/wip/rwbarton-tiny-tables
 * [new branch]            wip/simd                    -> origin/wip/simd
 * [new branch]            wip/smaller-minJumpTableSize -> origin/wip/smaller-minJumpTableSize
 * [new branch]            wip/spj-T13397              -> origin/wip/spj-T13397
 * [new branch]            wip/spj-float-in            -> origin/wip/spj-float-in
 * [new branch]            wip/spj-improvement         -> origin/wip/spj-improvement
 * [new branch]            wip/spj-solver-branch       -> origin/wip/spj-solver-branch
 * [new branch]            wip/spj-tc-branch           -> origin/wip/spj-tc-branch
 * [new branch]            wip/spj-tc-branch2          -> origin/wip/spj-tc-branch2
 * [new branch]            wip/spj-tc-branch3          -> origin/wip/spj-tc-branch3
 * [new branch]            wip/spj-temp                -> origin/wip/spj-temp
 * [new branch]            wip/spj-wildcard-refactor   -> origin/wip/spj-wildcard-refactor
 * [new branch]            wip/strictify-catches       -> origin/wip/strictify-catches
 * [new branch]            wip/t11715                  -> origin/wip/t11715
 * [new branch]            wip/tc-plugins              -> origin/wip/tc-plugins
 * [new branch]            wip/tc-plugins-amg          -> origin/wip/tc-plugins-amg
 * [new branch]            wip/tc/typeable-with-kinds  -> origin/wip/tc/typeable-with-kinds
 * [new branch]            wip/tdammers-7258           -> origin/wip/tdammers-7258
 * [new branch]            wip/tdammers/T14459         -> origin/wip/tdammers/T14459
 * [new branch]            wip/th-new                  -> origin/wip/th-new
 * [new branch]            wip/thomie                  -> origin/wip/thomie
 * [new branch]            wip/travis                  -> origin/wip/travis
 * [new branch]            wip/ttg-2017-10-13          -> origin/wip/ttg-2017-10-13
 * [new branch]            wip/ttg/2017-10-21          -> origin/wip/ttg/2017-10-21
 * [new branch]            wip/ttg3-2017-11-12         -> origin/wip/ttg3-2017-11-12
 * [new branch]            wip/ttg4-constraints-2017-11-13 -> origin/wip/ttg4-constraints-2017-11-13
 * [new branch]            wip/ttg5-data-2017-11-15    -> origin/wip/ttg5-data-2017-11-15
 * [new branch]            wip/ttg5-data-2017-11-17    -> origin/wip/ttg5-data-2017-11-17
 * [new branch]            wip/ttg5-data-one-file-2017-11-17 -> origin/wip/ttg5-data-one-file-2017-11-17
 * [new branch]            wip/ttg6-unrevert-2017-11-22 -> origin/wip/ttg6-unrevert-2017-11-22
 * [new branch]            wip/type-app                -> origin/wip/type-app
 * [new branch]            wip/use-O2                  -> origin/wip/use-O2
 * [new branch]            wip/zyla-T8095              -> origin/wip/zyla-T8095
 * [new tag]               04_07_06                    -> 04_07_06
 * [new tag]               07_04_06                    -> 07_04_06
 * [new tag]               2006-10-05                  -> 2006-10-05
 * [new tag]               2006-10-22                  -> 2006-10-22
 * [new tag]               2006-12-16                  -> 2006-12-16
 * [new tag]               2006-12-17                  -> 2006-12-17
 * [new tag]               2007-02-05                  -> 2007-02-05
 * [new tag]               2007-05-06                  -> 2007-05-06
 * [new tag]               2007-06-12                  -> 2007-06-12
 * [new tag]               2007-08-28                  -> 2007-08-28
 * [new tag]               2007-09-25                  -> 2007-09-25
 * [new tag]               2007-11-11                  -> 2007-11-11
 * [new tag]               2008-05-28                  -> 2008-05-28
 * [new tag]               2008-06-01                  -> 2008-06-01
 * [new tag]               2008-09-12                  -> 2008-09-12
 * [new tag]               2008-09-12_2                -> 2008-09-12_2
 * [new tag]               2009-03-13                  -> 2009-03-13
 * [new tag]               2009-06-25                  -> 2009-06-25
 * [new tag]               2009-11-15                  -> 2009-11-15
 * [new tag]               2010-03-16                  -> 2010-03-16
 * [new tag]               2010-11-18                  -> 2010-11-18
 * [new tag]               6_10_branch_has_been_forked -> 6_10_branch_has_been_forked
 * [new tag]               After_FC_branch_merge       -> After_FC_branch_merge
 * [new tag]               Approx_11550_changesets_converted -> Approx_11550_changesets_converted
 * [new tag]               Approx_2487_patches         -> Approx_2487_patches
 * [new tag]               Approximately_1000_patches_recorded -> Approximately_1000_patches_recorded
 * [new tag]               Approximately_9120_patches  -> Approximately_9120_patches
 * [new tag]               Before_FC_branch_merge      -> Before_FC_branch_merge
 * [new tag]               Before_cabalised-GHC        -> Before_cabalised-GHC
 * [new tag]               Before_type_family_merge    -> Before_type_family_merge
 * [new tag]               Initial_conversion_from_CVS_complete -> Initial_conversion_from_CVS_complete
 * [new tag]               final_switch_to_darcs,_this_repo_is_now_live -> final_switch_to_darcs,_this_repo_is_now_live
 * [new tag]               ghc-7.10.1-rc1              -> ghc-7.10.1-rc1
 * [new tag]               ghc-7.10.1-rc2              -> ghc-7.10.1-rc2
 * [new tag]               ghc-7.10.1-rc3              -> ghc-7.10.1-rc3
 * [new tag]               ghc-7.10.1-release          -> ghc-7.10.1-release
 * [new tag]               ghc-7.10.2-rc1              -> ghc-7.10.2-rc1
 * [new tag]               ghc-7.10.2-rc2              -> ghc-7.10.2-rc2
 * [new tag]               ghc-7.10.2-release          -> ghc-7.10.2-release
 * [new tag]               ghc-7.10.3-rc1              -> ghc-7.10.3-rc1
 * [new tag]               ghc-7.10.3-rc2              -> ghc-7.10.3-rc2
 * [new tag]               ghc-7.10.3-rc3              -> ghc-7.10.3-rc3
 * [new tag]               ghc-7.10.3-release          -> ghc-7.10.3-release
 * [new tag]               ghc-7.10.3a-release         -> ghc-7.10.3a-release
 * [new tag]               ghc-7.11-start              -> ghc-7.11-start
 * [new tag]               ghc-7.2.1-release           -> ghc-7.2.1-release
 * [new tag]               ghc-7.2.2-release           -> ghc-7.2.2-release
 * [new tag]               ghc-7.4.1-release           -> ghc-7.4.1-release
 * [new tag]               ghc-7.4.2-release           -> ghc-7.4.2-release
 * [new tag]               ghc-7.6.1-release           -> ghc-7.6.1-release
 * [new tag]               ghc-7.6.2-release           -> ghc-7.6.2-release
 * [new tag]               ghc-7.6.3-release           -> ghc-7.6.3-release
 * [new tag]               ghc-7.8.1-release           -> ghc-7.8.1-release
 * [new tag]               ghc-7.8.2-release           -> ghc-7.8.2-release
 * [new tag]               ghc-7.8.3-release           -> ghc-7.8.3-release
 * [new tag]               ghc-7.8.4-release           -> ghc-7.8.4-release
 * [new tag]               ghc-7.9-start               -> ghc-7.9-start
 * [new tag]               ghc-8.0.1-rc1               -> ghc-8.0.1-rc1
 * [new tag]               ghc-8.0.1-rc2               -> ghc-8.0.1-rc2
 * [new tag]               ghc-8.0.1-rc3               -> ghc-8.0.1-rc3
 * [new tag]               ghc-8.0.1-rc4               -> ghc-8.0.1-rc4
 * [new tag]               ghc-8.0.1-release           -> ghc-8.0.1-release
 * [new tag]               ghc-8.0.2-rc1               -> ghc-8.0.2-rc1
 * [new tag]               ghc-8.0.2-rc2               -> ghc-8.0.2-rc2
 * [new tag]               ghc-8.0.2-release           -> ghc-8.0.2-release
 * [new tag]               ghc-8.1-start               -> ghc-8.1-start
 * [new tag]               ghc-8.2.1-rc1               -> ghc-8.2.1-rc1
 * [new tag]               ghc-8.2.1-rc2               -> ghc-8.2.1-rc2
 * [new tag]               ghc-8.2.1-rc3               -> ghc-8.2.1-rc3
 * [new tag]               ghc-8.2.1-release           -> ghc-8.2.1-release
 * [new tag]               ghc-8.2.2-rc1               -> ghc-8.2.2-rc1
 * [new tag]               ghc-8.2.2-rc2               -> ghc-8.2.2-rc2
 * [new tag]               ghc-8.2.2-rc3               -> ghc-8.2.2-rc3
 * [new tag]               ghc-8.2.2-release           -> ghc-8.2.2-release
 * [new tag]               ghc-8.3-start               -> ghc-8.3-start
 * [new tag]               ghc-8.4.1-alpha1            -> ghc-8.4.1-alpha1
 * [new tag]               ghc-8.4.1-alpha2            -> ghc-8.4.1-alpha2
 * [new tag]               ghc-8.5-start               -> ghc-8.5-start
 * [new tag]               ghc-darcs-git-switchover    -> ghc-darcs-git-switchover
Switched to a new branch 'fetchgit'
Submodule '.arc-linters/arcanist-external-json-linter' (git://git.haskell.org/arcanist-external-json-linter.git) registered for path '.arc-linters/arcanist-external-json-linter'
Submodule 'hadrian' (git://git.haskell.org/hadrian.git) registered for path 'hadrian'
Submodule 'libffi-tarballs' (git://git.haskell.org/libffi-tarballs.git) registered for path 'libffi-tarballs'
Submodule 'libraries/Cabal' (git://git.haskell.org/packages/Cabal.git) registered for path 'libraries/Cabal'
Submodule 'libraries/Win32' (git://git.haskell.org/packages/Win32.git) registered for path 'libraries/Win32'
Submodule 'libraries/array' (git://git.haskell.org/packages/array.git) registered for path 'libraries/array'
Submodule 'libraries/binary' (git://git.haskell.org/packages/binary.git) registered for path 'libraries/binary'
Submodule 'libraries/bytestring' (git://git.haskell.org/packages/bytestring.git) registered for path 'libraries/bytestring'
Submodule 'libraries/containers' (git://git.haskell.org/packages/containers.git) registered for path 'libraries/containers'
Submodule 'libraries/deepseq' (git://git.haskell.org/packages/deepseq.git) registered for path 'libraries/deepseq'
Submodule 'libraries/directory' (git://git.haskell.org/packages/directory.git) registered for path 'libraries/directory'
Submodule 'libraries/dph' (git://git.haskell.org/packages/dph.git) registered for path 'libraries/dph'
Submodule 'libraries/filepath' (git://git.haskell.org/packages/filepath.git) registered for path 'libraries/filepath'
Submodule 'libraries/haskeline' (git://git.haskell.org/packages/haskeline.git) registered for path 'libraries/haskeline'
Submodule 'libraries/hpc' (git://git.haskell.org/packages/hpc.git) registered for path 'libraries/hpc'
Submodule 'gmp-tarballs' (git://git.haskell.org/gmp-tarballs.git) registered for path 'libraries/integer-gmp/gmp/gmp-tarballs'
Submodule 'libraries/mtl' (git://git.haskell.org/packages/mtl.git) registered for path 'libraries/mtl'
Submodule 'libraries/parallel' (git://git.haskell.org/packages/parallel.git) registered for path 'libraries/parallel'
Submodule 'libraries/parsec' (git://git.haskell.org/packages/parsec.git) registered for path 'libraries/parsec'
Submodule 'libraries/pretty' (git://git.haskell.org/packages/pretty.git) registered for path 'libraries/pretty'
Submodule 'libraries/primitive' (git://git.haskell.org/packages/primitive.git) registered for path 'libraries/primitive'
Submodule 'libraries/process' (git://git.haskell.org/packages/process.git) registered for path 'libraries/process'
Submodule 'libraries/random' (git://git.haskell.org/packages/random.git) registered for path 'libraries/random'
Submodule 'libraries/stm' (git://git.haskell.org/packages/stm.git) registered for path 'libraries/stm'
Submodule 'libraries/terminfo' (git://git.haskell.org/packages/terminfo.git) registered for path 'libraries/terminfo'
Submodule 'libraries/text' (git://git.haskell.org/packages/text.git) registered for path 'libraries/text'
Submodule 'libraries/time' (git://git.haskell.org/packages/time.git) registered for path 'libraries/time'
Submodule 'libraries/transformers' (git://git.haskell.org/packages/transformers.git) registered for path 'libraries/transformers'
Submodule 'libraries/unix' (git://git.haskell.org/packages/unix.git) registered for path 'libraries/unix'
Submodule 'libraries/vector' (git://git.haskell.org/packages/vector.git) registered for path 'libraries/vector'
Submodule 'libraries/xhtml' (git://git.haskell.org/packages/xhtml.git) registered for path 'libraries/xhtml'
Submodule 'nofib' (git://git.haskell.org/nofib.git) registered for path 'nofib'
Submodule 'utils/haddock' (git://git.haskell.org/haddock.git) registered for path 'utils/haddock'
Submodule 'utils/hsc2hs' (git://git.haskell.org/hsc2hs.git) registered for path 'utils/hsc2hs'
Initialized empty Git repository in /nix/store/5z011gagfisjknv0zdcs4cq1hf94zfv6-ghc-61db0b8/.arc-linters/arcanist-external-json-linter/.git/
fatal: The remote end hung up unexpectedly
fatal: protocol error: bad pack header
remote: Counting objects: 31, done.        
remote: Compressing objects: 100% (30/30), done.        
remote: Total 31 (delta 12), reused 0 (delta 0)        
From git://git.haskell.org/arcanist-external-json-linter
 * [new branch]      master     -> origin/master
fatal: '57f9e4e8eb422dcb72f02f07767a1ed5969baab' is not a commit and a branch 'fetchgit' cannot be created from it
Unable to checkout 57f9e4e8eb422dcb72f02f07767a1ed5969baab from git://git.haskell.org/arcanist-external-json-linter.git.
builder for ‘/nix/store/lzgaghhba2ci567yyfsg41392pj1rb6m-ghc-61db0b8.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/i00rsspj8fkpx37jkr18g9hvl2lsbvim-ghc-8.4.20180122.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/i00rsspj8fkpx37jkr18g9hvl2lsbvim-ghc-8.4.20180122.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

**Comment by dtzWill** [comment]
This is fixed by #34232-- our git fetcher is currently broken re:submodules.

Please give that a try (or wait until it is merged) and update us with your findings.

**Comment by mpickering** [comment]
Thanks, that commit fixed it.
